corehq/apps/userreports/dbaccessors.py

```diff
@@ -50,3 +50,16 @@ def get_all_report_configs():
 def delete_all_report_configs():
     from corehq.apps.userreports.models import ReportConfiguration
     delete_all_docs_by_doc_type(ReportConfiguration.get_db(), ('ReportConfiguration',))
+
+
+def _get_all_data_sources():
+    from corehq.apps.userreports.models import DataSourceConfiguration
+    return DataSourceConfiguration.view(
+        'userreports/data_sources_by_build_info',
+        reduce=False,
+        include_docs=True
+    )
```

Comment:
we could even.... add a couch view to just pull the index names. :smiling_imp: 

it is just the meta db so shouldn't be expensive